File: Calculator/ios-native/CalculatorHub/Calculators/SimpleInterestCalculator.swift
import Foundation

struct SimpleInterestResult {
    let interest: Double
    let totalAmount: Double
}

class SimpleInterestCalculator {
    func calculate(principal: Double, annualRate: Double, timeYears: Double) -> SimpleInterestResult {
        let rate = annualRate / 100.0
        let interest = principal * rate * timeYears
        let total = principal + interest
        
        return SimpleInterestResult(
            interest: interest,
            totalAmount: total
        )
    }
}

File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/SimpleInterestCalculator.kt
package com.qaantum.calculatorhub.calculators

data class SimpleInterestResult(
    val interest: Double,
    val totalAmount: Double
)

class SimpleInterestCalculator {
    fun calculate(
        principal: Double,
        annualRate: Double,
        timeYears: Double
    ): SimpleInterestResult {
        val rate = annualRate / 100.0
        val interest = principal * rate * timeYears
        val total = principal + interest
        
        return SimpleInterestResult(
            interest = interest,
            totalAmount = total
        )
    }
}
